packages/liquidity-widgets/src/stores/widget.tsx

```diff
@@ -0,0 +1,274 @@
+        maxTick: nearestUsableTick(MAX_TICK, pool.positionInfo.tickSpacing),
+      };
+
+      set({ pool: p });
+
+      if (positionId !== undefined) {
+        const contract = DexInfos[poolType].nftManagerContract;
+        const contractAddress =
+          typeof contract === "string" ? contract : contract[chainId];
+        if (!contractAddress) {
+          set({
+            errorMsg: `Pool type ${poolType} is not supported in chainId: ${chainId}`,
+          });
+          return;
+        }
+        // Function signature and encoded token ID
+        const functionSignature = "positions(uint256)";
+        const selector = getFunctionSelector(functionSignature);
+        const encodedTokenId = encodeUint256(BigInt(positionId));
+
+        const data = `0x${selector}${encodedTokenId}`;
+
+        // JSON-RPC payload
+        const payload = {
+          jsonrpc: "2.0",
+          method: "eth_call",
+          params: [
+            {
+              to: contractAddress,
+              data: data,
+            },
+            "latest",
+          ],
+          id: 1,
+        };
+
+        // Send JSON-RPC request via fetch
+        const response = await fetch(NetworkInfo[chainId].defaultRpc, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(payload),
+        });
+
+        const { result, error } = await response.json();
+
+        if (result && result !== "0x") {
+          const data = decodePosition(result);
+
+          const { amount0, amount1 } = getPositionAmounts(
+            p.tick,
+            data.tickLower,
+            data.tickUpper,
+            BigInt(p.sqrtPriceX96),
+            data.liquidity
+          );
+
+          set({
+            position: {
+              id: +positionId,
+              poolType,
+              liquidity: data.liquidity,
+              tickLower: data.tickLower,
+              tickUpper: data.tickUpper,
+              amount0,
+              amount1,
+            },
+          });
+          return;
+        }
+
+        set({ errorMsg: error.message || "Position not found" });
+      }
+    },
+  }));
+};
+
+type WidgetStore = ReturnType<typeof createWidgetStore>;
+
+const WidgetContext = createContext<WidgetStore | null>(null);
+
+export function WidgetProvider({ children, ...props }: WidgetProviderProps) {
+  const store = useRef(createWidgetStore(props)).current;
+
+  const { fetchPrices } = useTokenPrices({
+    addresses: [],
+    chainId: store.getState().chainId,
+  });
+
+  useEffect(() => {
+    // get Pool and position then update store here
+    store.getState().getPool(fetchPrices);
+    const i = setInterval(() => {
+      store.getState().getPool(fetchPrices);
+    }, 15_000);
+    return () => clearInterval(i);
+  }, []);
+
+  return (
+    <WidgetContext.Provider value={store}>{children}</WidgetContext.Provider>
```

Comment:
@viet-nv We don't need a context provider to pass the store to the whole app. `zustand` supports us with just a hook called from every component that requires the state. Zustand team even mentioned that if we don't have a particular purpose, don't pass the store through the React context. https://github.com/pmndrs/zustand?tab=readme-ov-file#react-context

I will change it later in my PR, so It's just a comment.

Reply:
Maybe you misunderstand. `The recommended method available since v4 is to use the vanilla store.`
in this case, I want  to initialize your store with props from a component, this is sample code from their docs